### ISSUE TITLE
`takeSorted` ignores comparator

### ISSUE THREAD

**Issue by stevenwdv** [iter-tools/iter-tools#444]
It seems to me that the `takeSorted` function does not use the comparator function at all:
```ts
console.log([...takeSorted((a, b) => b - a, [1, 3, 2])]);
console.log([...takeSorted((a, b) => a - b, [1, 3, 2])]);
console.log([...takeSorted(3, (a, b) => b - a, [1, 3, 2])]);
console.log([...takeSorted(3, (a, b) => a - b, [1, 3, 2])]);
console.log([...takeSorted<number>(3, (a, b) => b - a)([1, 3, 2])]);
console.log([...takeSorted<number>(3, (a, b) => a - b)([1, 3, 2])]);
```
These all print `[ 1, 2, 3 ]`.

**Comment by conartist6** [comment]
Which version of iter-tools are you using?

**Comment by conartist6** [comment]
I just stuck the code you shared into the test suite for `v7.1.3` and the results look fine, i.e. `[3, 2, 1]` when comparator is `b - a`.

**Comment by conartist6** [comment]
I don't really plan to maintain the 6.x line -- that's the era before I took over maintainership of the project. There are so many bugs in there that attempting to release patch fixes for all of them would just be repeating the whole process I went through to the create 7.x, including a complete overhaul of the test suite to gain full coverage. The test suite rewrite shook out countless bugs, some of them logged here, some not.

**Comment by conartist6** [comment]
Wait wait. I just remembered I did see a repro in runkit though...

https://runkit.com/conartist6/614261a0f64616000808cc50

**Comment by conartist6** [comment]
Ooh OK I take it all back. I guess my test coverage still isn't quite what it could be. Looks like an issue in `validateArgs` left over from rework I did on my currying helpers.

**Comment by conartist6** [comment]
Fixed in `v7.1.4`.

I hope you don't mind me arguing with myself here. It beats waiting days, weeks, months for a fix right?? Ugh, I'm just bitter from my experience reporting issues that never get fixed. Anyway, if you like iter-tools recommend it to your friends or coworkers or whoever.

**Comment by stevenwdv** [comment]
Don't worry about it, I can always appreciate some progress updates! I'm glad you fixed it :) (sorry for the late reply)
Btw, this lib will be even nicer when we get the [pipeline operator](https://github.com/tc39/proposal-pipeline-operator). Although you may also get a '[rival](https://github.com/tc39/proposal-iterator-helpers)'.
